### PR TITLE
Add support for 0rtt connection resumption to DoH and DoQ listeners. 

### PR DESCRIPTION
**Pull request by LeonardWalter**
Add support for 0rtt connection resumption to DoH and DoQ listeners.

### CODE REVIEW

File: doqlistener.go [hunk 1]
```diff
@@ -128,31 +126,31 @@ func (s DoQListener) handleConnection(connection quic.Connection) {
...
 	// DNS over QUIC uses one stream per query/response.
-	defer stream.Close()
 	s.metrics.stream.Add(1)
 
 	// DoQ requires a length prefix, like TCP
 	var length uint16
 	if err := binary.Read(stream, binary.BigEndian, &length); err != nil {
 		s.metrics.err.Add("read", 1)
 		log.WithError(err).Error("failed to read query")
+		CloseConn(connection, log)
```
Comment (folbricht) [inline doqlistener.go:153]: Not a fan of closing the connection in multiple places, including from within the stream. One effect of doing it here will be that `connection.AcceptStream(ctx)` above will return an error, which will then call `CloseConn` a second time. It should write `"closing connection"` twice in the logs too.

The prior implementation only closed it in one place. Is that not working? If a stream fails, I'd expect `AcceptStream()` to also fail, which then closes the connection.

**Review by folbricht — COMMENTED** [review]
Thank you. Let's make sure that closing the connection in multiple places is actually necessary.

File: doqlistener.go (hunk 1 above)
Comment (LeonardWalter) [reply doqlistener.go]: I agree. I did remove the changes and updated the PR

**Comment by LeonardWalter** [comment]
There is currently still a problem when the server restarts and the client still holds an old session ticket. I think the client should automatically retry without the session ticket after it failed to (re)open the connection. Will work on this now

**Pull request merged**